sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryQuerySource.java

```diff
@@ -96,62 +90,35 @@ private BigQueryQuerySource(
     this.query = checkNotNull(query, "query");
     this.flattenResults = checkNotNull(flattenResults, "flattenResults");
     this.useLegacySql = checkNotNull(useLegacySql, "useLegacySql");
-    this.dryRunJobStats = new AtomicReference<>();
     this.priority = priority;
     this.location = location;
     this.kmsKey = kmsKey;
+    queryHelperReference = new AtomicReference<>();
+  }
+
+  /**
+   * Since the query helper reference is declared as transient, neither the AtomicReference nor the
+   * structure it refers to are persisted across serialization boundaries. The code below is
+   * resilient to the QueryHelper object disappearing in between method calls, but the reference
```

Comment:
There's some transient state associated with the helper -- specifically, in order to avoid unnecessarily doing multiple dry runs of the query, we keep around the resulting job statistics object using the same pattern (https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryQuerySource.java#L79). I can make the change you're recommending, though.

Reply:
Done.